operator/pkg/ciliumendpointslice/manager.go

```diff
@@ -282,18 +285,22 @@ func (c *cesMgr) getCESCopyFromCache(cesName string) (*cilium_v2.CiliumEndpointS
 // CES object or updating an existing CES object.
 func (c *cesMgr) InsertCEPInCache(cep *cilium_v2.CoreCiliumEndpoint, ns string) string {
 	log.WithFields(logrus.Fields{
-		"CEPName": GetCEPNameFromCCEP(cep, ns),
+		logfields.CEPName: GetCEPNameFromCCEP(cep, ns),
 	}).Debug("Insert CEP in local cache")
 
 	// check the given cep is already exists in any of the CES.
 	// if yes, Update a ces with the given cep object.
 	if cesName, exists := c.desiredCESs.getCESName(GetCEPNameFromCCEP(cep, ns)); exists {
-		ces := c.desiredCESs.getCESTrackerOnly(cesName)
-		// add a cep into the ces
-		c.addCEPtoCES(cep, ces)
-		return cesName
+		if ces, ok := c.desiredCESs.getCESTracker(cesName); ok {
+			// add a cep into the ces
+			c.addCEPtoCES(cep, ces)
+			return cesName
+		} else {
+			log.WithFields(logrus.Fields{
+				logfields.CESName: cesName,
+			}).Debug("Attempted to InsertCEPInCache non-existent cesName , skipping.")
```

Comment:
Suggest:
"CEP to CES mapping incorrect - missing CESName"

and add both CEP and CES names to the log fields.